dtfs-central-api/src/services/utilisation-report-data-validator/index.test.ts

```diff
@@ -0,0 +1,140 @@
+import { UTILISATION_REPORT_HEADERS } from '@ukef/dtfs2-common';
+import { validateUtilisationReportCsvHeaders, validateUtilisationReportCsvCellData, getHeaderIsMissingErrorMessage } from '.';
+import {
+  generateUkefFacilityIdError,
+  generateBaseCurrencyError,
+  generateFacilityUtilisationError,
+  generatePaymentCurrencyError,
+  generatePaymentExchangeRateError,
+} from './utilisation-report-cell-validators';
+
+jest.mock('./utilisation-report-cell-validators', () => ({
+  generateUkefFacilityIdError: jest.fn(),
+  generateBaseCurrencyError: jest.fn(),
+  generateFacilityUtilisationError: jest.fn(),
+  generatePaymentCurrencyError: jest.fn(),
+  generatePaymentExchangeRateError: jest.fn(),
+  generateTotalFeesAccruedCurrencyError: jest.fn(),
+  generateTotalFeesAccruedExchangeRateError: jest.fn(),
+}));
+
+describe('utilisation-report-validator', () => {
+  describe('getHeaderIsMissingErrorMessage', () => {
+    it("returns the passed in string followed by the text 'header is missing or spelt incorrectly'", () => {
+      // Arrange
+      const header = 'some string';
+
+      // Act
+      const error = getHeaderIsMissingErrorMessage(header);
+
+      // Assert
+      expect(error).toEqual(`${header} header is missing or spelt incorrectly`);
+    });
+  });
+
+  describe('validateUtilisationReportCsvHeaders', () => {
+    it('returns an error if a header is missing', () => {
+      const csvDataRowWithMissingHeader = {
+        'bank facility reference': { value: 'Britannia Energy GEF', column: 'A', row: 1 },
```

Comment:
are any of these stored in constants?

Reply:
I've reworked the tests after a similar comment from Tony to try and make them better.